File: TodoSpa/src/types/todo.ts
export interface Todo {
    id: number;
    title: string;
    isComplete: boolean;
    createdBy: number;
    modifiedBy?: number;
    createdAt: string;
    modifiedAt?: string;
    isDeleted: boolean;
}

export interface CreateTodoRequest {
    title: string;
}

export interface UpdateTodoRequest {
    title?: string;
    isComplete?: boolean;
}

File: TodoApi/Models/Todo.cs
namespace TodoApi.Models;

public class Todo
{
    public int Id { get; set; }
    public string? Title { get; set; }
    public bool IsComplete { get; set; }
    public int CreatedBy { get; set; }
    public int? ModifiedBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ModifiedAt { get; set; }
    public bool IsDeleted { get; set; } = false;
}
